### ISSUE TITLE
what kind of upload scripts can i run

### ISSUE THREAD

**Issue by a7medo778** [ampache/ampache#1587]
hi guys
so what kind of upload scripts can i run ?
and how can i fine tune the permissions to allow the uploader to only manage his uploads?

and finally (this is a general question), how can i save the thumbs of the artists/albums in their folders? @wagnered 

thanks in advanced 
Ahmad

**Comment by a7medo778** [comment]
1 additional question
can we set additional tags for the uploads ? right now the uploader only choses the artist and album
can i force something like the genre to be applied to the tags aswell ?

**Comment by wagnered** [comment]
@a7medo778 

> so what kind of upload scripts can i run ?

I haven't looked too closely at this feature, but I'm assuming php scripts.

> how can i fine tune the permissions to allow the uploader to only manage his uploads?

Possibly disabling sharing in the Admin->options page.

> how can i save the thumbs of the artists/albums in their folders? 

If I understand correctly, remove the comment (;) symbol from album_art_store_disk = "false" in ampache.cfg.php and change to "true".

>can we set additional tags for the uploads ?

I currently can't get the upload to work correctly, but if my memory serves me correctly, sorry to say, no.  I also think this feature should be more flexible: Should be able to set new artist name, folder and other major tags.

I'm marking ths as an enhancement to at least look into.

**Comment by a7medo778** [comment]
thanks @wagnered for the answers, much appreciate it

i also look forward for this feature to be more mature as it really a unique ampache feature among all the other subsonic class platforms

**Comment by a7medo778** [comment]
hey @wagnered i think i made it work to some extent
1 question though, how can i increase the uploade size limit from 8mb ? its not in the config file or the settings

**Comment by wagnered** [comment]
@a7medo778,

Check "upload_max_filesize" in your php.ini config file, which on Fedora
defaults to 2M and also "max_file_uploads" which on Fedora defaults to
20 which would limit the total size of uploads at any given time.


73 From  AA1AD
Ernie D
Federal Way, WA | CN87ug

**Comment by a7medo778** [comment]
@wagnered
"max_upload_size | 1048576 in php debug"

**Comment by wagnered** [comment]
Did you check "max_upload_size" in ampache.cfg.php?

73 From  AA1AD
Ernie D
Federal Way, WA | CN87ug

**Comment by a7medo778** [comment]
Yes its the same value

Regards
Ahmed

**Comment by a7medo778** [comment]
<img width="393" alt="screen shot 2017-08-26 at 1 16 22 am" src="https://user-images.githubusercontent.com/1003222/29734678-452c7702-89fc-11e7-90ef-c87be74cbdd2.png">

**Comment by a7medo778** [comment]
@wagnered 
+ after enabling "save on disk" option for the art, all pictures became unavailable until i enabled the second option which had a path
now the art isn't being save in the same folder as the mp3's but in that folder, i had to specify

i am sorry i am asking too many questions but my prject is going live in less than a month

**Comment by wagnered** [comment]
@a7medo778,

I don't understand what your question is.  Are you referring to uploaded
files? Why can't the art be saved to the database? The config switch
"write_id3_art will allow this.

Did you get the upload size problem solved?

73 From  AA1AD
Ernie D
Federal Way, WA | CN87ug

**Comment by a7medo778** [comment]
No its still limited to 8MB (pleeeease help in that in tried every everything 😩)

I am trying to get the art saved in the same folder as the mp3's, so when scanned in the future it will all be there

Regards
Ahmed

**Comment by wagnered** [comment]
OK, The second image in your post yesterday wasn't in the message I was forwarded. NOw I understand.

Check "post_max_size" in the php.ini file.  Because Ampache uses _POST to upload files, it uses this to limit the upload size.

**Comment by a7medo778** [comment]
<img width="561" alt="screen shot 2017-08-27 at 1 09 11 am" src="https://user-images.githubusercontent.com/1003222/29745494-654f5676-8ac4-11e7-88eb-f2404b4ecccb.png">
chnaged it to 80M but still seeing that 8 M limit on the upload screen...

do i need to restart Apache or something?

**Comment by wagnered** [comment]
My apologies, I should have suggested doing that.  Of course, any time
the php.ini is modified the Apache server must be restarted so the php
engine can read the changes in the php.ini file.


73 From  AA1AD
Ernie D
Federal Way, WA | CN87ug

**Comment by a7medo778** [comment]
@wagnered it didnt work :/
can you verify that this is the method from your end, if its working for you then i should dig deeper into apache config

**Comment by wagnered** [comment]
> can you verify that this is the method from your end.

If you are referring to restarting Apache (httpd?) Yes, it must be restarted.  Are  you sure that you did restart the process?

How about giving us more information about your setup. See the [wiki](https://github.com/ampache/ampache/wiki/Issues) on what information to report.

As I stated earlier, the image of your upload page clued me in as to what was going on.  I traced the code and discovered that it's either  `upload_max_filesize` or `post_max_size` which are set to 2M and 8M respectively in a fresh php.ini file on Fedora.

If you beliee that you successfully restarted Apache (httpd), run the command `php --ini` from a command line and look for "Loaded Configuration File:" in the output.  There is a possibility that the php.ini file that you modified is not the one that php iis actually using.

IT's not a problem of Apache exactly, but Apache runs the php process which must be restarted when php.ini changes.
